common/flatpak-dir.c

```diff
@@ -8662,19 +8668,20 @@ flatpak_dir_deploy_update (FlatpakDir        *self,
   old_origin = flatpak_deploy_data_get_origin (old_deploy_data);
   old_subpaths = flatpak_deploy_data_get_subpaths (old_deploy_data);
 
-  previous_ids = g_strdupv ((char **) flatpak_deploy_data_get_previous_ids (old_deploy_data, NULL));
+  previous_ids = flatpak_deploy_data_get_previous_ids (old_deploy_data, NULL);
+  previous_ids_owned = g_strdupv ((char **) previous_ids);
   if (opt_previous_ids)
     {
-      g_auto(GStrv) old_previous_ids = previous_ids;
-      previous_ids = flatpak_strv_merge (old_previous_ids, (char **) opt_previous_ids);
+      previous_ids_owned = flatpak_strv_merge ((char **) previous_ids, (char **) opt_previous_ids);
+      g_clear_pointer (&previous_ids, g_free);
```

Comment:
Why is this needed whe previous_ids is already a autofree?

Reply:
It's not strictly needed but it seems more robust this way so that someone can't inadvertently add code to use the now-outdated value.